YAFCmodel/Data/DataUtils.cs

```diff
@@ -411,7 +411,7 @@ public class FavouritesComparer<T> : IComparer<T> where T : FactorioObject {
             var multiplier = unit == UnitOfMeasure.Megawatt ? 1e6f : 1f;
             amount = 0;
             foreach (var c in str) {
-                if (c >= '0' && c <= '9' || c == '.' || c == '-' || c == 'e')
+                if (c is >= '0' and <= '9' or '.' or '-' or 'e')
```

Comment:
Personally, I am not a big fan of this, I do not find it clear(er), but I can adjust


IDE will help me getting this syntax right :smile:

Reply:
Agreed. I can justify this pattern matching [here](https://github.com/have-fun-was-taken/yafc-ce/pull/64/files/fe429f64dba69c1c8fd46a1c064c418b81dc71c9..7e869750f5da2a87812a6af8e89fd91d012ba250#diff-574901d659ae628ac0805d2c373c7a94d404780ed54a5e7d92988e8c4ad272bbR281), but not at the line you mentioned.

Reply:
Yes those lines are very nice and readable with this change. For them I really do like it a lot.

Reply:
The problem is that I don't see a setting to turn it off only for intense usages, and the used operators are largely the same. With parentheses enabled, it's sort-of readable: 
```
if (c is (>= '0' and <= '9') or '.' or '-' or 'e')
```
Therefore, I'm leaning towards leaving IDE0078 enabled because other use cases improve readability.

Reply:
agreed, this is probably an outlier